### PR TITLE
EM::Connection#watch_only?

### PR DESCRIPTION
**Pull request by genki**
This PR provides a way to test if a connection is watch_only.

### CODE REVIEW

File: ext/cmain.cpp [hunk 1]
```diff
@@ -360,6 +360,19 @@ extern "C" void evma_unwatch_pid (const uintptr_t sig)
...
 
+/*****************
+evma_is_watch_only
+*****************/
+
+extern "C" int evma_is_watch_only (const uintptr_t binding)
+{
+	EventableDescriptor *cd = dynamic_cast <EventableDescriptor*> (Bindable_t::GetObject (binding));
+	if (cd)
+    return cd->IsWatchOnly() ? 1 : 0;
```
Comment (sodabrew) [inline ext/cmain.cpp]: This looks like a spaces vs. tabs mismatch.

**Comment by genki** [comment]
My use case is to list up all EM connections and close them safely while a sort of finalization process.
Current implementation throws C-level exception (hence causes a crash) if the connection to be closed is the watchonly, even there is no way to check if it is watchonly.
By using this PR, we can check it before calling close to avoid a panic.

File: ext/cmain.cpp (hunk 1 above)
Comment (genki) [inline ext/cmain.cpp]: Changed spaces to TAB.

**Comment by sodabrew** [comment]
On what platform does it crash when closing the (file-?)watch? If it's Linux only then this is actually a known issue that is fixed in 1.2.2 and I'll upload it in just a minute.

**Comment by sodabrew** [comment]
You might be experiencing issue #512. Change #757 in the just-released EventMachine 1.2.2 catches the C++ exception and raises a rescueable Ruby exception. Please give it a try!

**Comment by genki** [comment]
@olegshaldybin Thanks, I will try.

**Comment by sodabrew** [comment]
Let me know if EventMachine 1.2.2 helped your situation @genki? If this new API would still be helpful to your program, I want to make sure to add the API to docs and other reactors as well.

**Comment by genki** [comment]
@sodabrew Sorry for delayed response.
Your fix looks great but  there is no chance to know which method should be called either `close` or `stop_watching`, for the listed connection objects without using `watch_only?`

**Pull request merged**